### PR TITLE
Reorganizes post options in Settings.

### PR DESCRIPTION
**Pull request by kurzee**
Resolves the conversation and issues raised in https://github.com/wordpress-mobile/WordPress-iOS/issues/4756, https://github.com/wordpress-mobile/WordPress-iOS/pull/4757 and https://github.com/wordpress-mobile/WordPress-iOS/pull/4766.

After much thought and conversation with @aerych. The best approach moving forward allows for both admin and non-admin users of XML-RPC sites to toggle default post options under a new section in settings titled "This Device".

This PR also moves "Geotagging" under "This Device" for all users, XML-RPC and REST.

REST users only have default post options available if they are admins of the connected site and under the section "Writing". Non-admins will instead default to what the admin has selected for the site. (This is in parity with Calypso on the web)

REST Testing:
1. Open Site Settings for a site connected via the REST-API and as an admin user.
2. Observe the available default category, post format and related posts options under "Writing".
3. Observe the available Geotagging option under "This Device".
4. Selecting any of the available options updates both in-app and on the site.

REST Testing:
1. Open Site Settings for a site connected via the REST-API and as a non-admin user.
2. Observe no available "Writing" section.
3. Observe the available Geotagging option under "This Device".

XML-RPC Testing:
1. Open Site Settings for a site connected via the XML-RPC as admin or non-admin user.
2. Observe the availability for the default category, post format options under "This Device".
3. Observe the available Geotagging option under "This Device" as well.

@aerych please review and @mattmiklic on the text review, one more time. 😃Thanks so much for your input guys.

### CODE REVIEW

**Comment by mattmiklic** [comment]
Tested again, everything's looking good to me.

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m [hunk 1]
```diff
@@ -366,6 +382,28 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForWritingSettingsAt
...
+        case (SiteSettingsDeviceDefaultCategory):{
+            [self configureDefaultCategoryCell];
+            return self.defaultCategoryCell;
+        }
+            break;
+        case (SiteSettingsDeviceDefaultPostFormat):{
+            [self configureDefaultPostFormatCell];
+            return self.defaultPostFormatCell;
+        }
+            break;
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: This is nitpicky code style stuff, but for consistency within the file maybe remove the parenthesis in the `case` statements and remove the `break`s since each case returns?

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m [hunk 2]
```diff
@@ -726,7 +777,20 @@ - (void)tableView:(UITableView *)tableView didSelectInWritingSectionRow:(NSInteg
             [self showRelatedPostsSettings];
         }
         break;
+    }
+}
 
+- (void)tableView:(UITableView *)tableView didSelectInDeviceSectionRow:(NSInteger)row
+{
+    switch (row) {
+        case SiteSettingsDeviceDefaultCategory:{
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: No need for the `{` `}` here, but no harm either.

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m (hunk 1 above)
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: The style consistency of the switch cases and breaks in this file are all over the place. I'm going to leave this alone for now as it would require touching a lot more code than the context of this PR.

**Comment by aerych** [comment]
Hiya @kurzee :) 

I left some super nitpicky code style comments.  I tested settings on wpcom blogs with both admin and non-admin users, and a self-hosted test blog via XML-RPC. Everything worked as described. (Yay!) 

I *did* notice an edge case that is probably not a blocker (existing issue I think) but I want to make a note of it here.  Steps to reproduce:
- Start with a fresh install and sign into a self-hosted blog that is connected to Jetpack. 
- Go to settings and confirm that there are options under This Device for geotagging, default post format and default category. 
- Go to the Me tab and sign into the wpcom account that Jetpack is connected to. 
- Switch back to the My Sites tab.  It should still be showing the settings screen, however the options on the screen have not updated to reflect a site with a REST API connection. The options for default post format and default category are missing from the Writing section and the Remove Site option is still visible. 
- Tap the back button to return to the site detail.
- Tap to view settings again.  Everything is fixed and appears as it should. 

I recommend we open an new issue for the wonky behavior rather than draw out this patch any further. :)

Once the code style comments are addressed I say :shipit:!  Nice work yo!

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m (hunk 1 above)
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: Fair enough :)

**Comment by kurzee** [comment]
Merging. Thanks @aerych! Will create a separate issue for the edge case mentioned above.
